nodes.py

```
@@ -60,6 +60,23 @@ def replace_format(match):
+    """
+    Replace %counter<padding> patterns with formatted counter.
+    Example: %counter<03> with counter=1 becomes 001
+    """
+    pattern = r'%counter<([0-9]+)>'
+    def replace_format(match):
+        padding_str = match.group(1)
+        try:
+            # Create format string like "{:03d}"
+            fmt = "{:0" + padding_str + "d}"
+            return fmt.format(counter)
+        except:
```

Ruff (0.15.12)

[error] Do not use bare `except`

(E722)